t/onfido.t

```diff
@@ -190,4 +191,51 @@ is($token->{referrer}, 'https://*.example.com/example_page/*', 'referrer is ok i
 # applicant delete
 lives_ok { $onfido->applicant_delete(applicant_id => $app->id)->get } "delete ok";
 
+# ratelimit
+# clear rate limiting
+$loop->add(
+    $onfido = WebService::Async::Onfido->new(
+        token    => 'test_token',
+        base_uri => 'http://localhost:3000',
+        requests_per_minute => 5,
+    ));
+
+
+for (1..5){
+    ok(!$onfido->is_rate_limited, "not limited yet");
+    my $result = $onfido->rate_limiting;
+    ok($result->is_ready, 'all results are ready at first because they are in the rate limit');
+}
+my @results;
+for (1..10){
+    ok($onfido->is_rate_limited, "is rate_limited now");
+    my $result = $onfido->rate_limiting;
+    ok(!$result->is_ready, 'all results are not ready now because they are out of rate limit');
+    push @results, $result;
+}
+
+# advance 1st interval
+set_relative_time(60);
+# trigger loop delay_future
+$onfido->loop->loop_once(0);
+ok($onfido->is_rate_limited, "still rate_limited again");
+for (1..5){
+    my $result = shift @results;
+    ok($result->is_ready, 'now the first 5 future should be ready because they are in the rate limit now');
+}
+ok($onfido->is_rate_limited, "here still rate_limited");
+for (1..5){
+    my $result = shift @results;
+    ok(!$result->is_ready, 'now the last 5 futures should not be ready because they are still out of rate limit');
+}
+
+# advance 2nd interval
+set_relative_time(60*2);
+# trigger loop delay_future
+$onfido->loop->loop_once(0);
+# advance 3rd interval
```

Comment:
Should we check after this interval that api client still rate limited?

Reply:
We did at line 239. I guess you mean after 2nd interval? let me add one.